OvmfPkg/TdxDxe/X64/ApRunLoop.nasm

```diff
@@ -125,7 +125,7 @@ AsmRelocateApResetVector:
 
 .prepareStack:
     ; The stack can then be used to switch from long mode to compatibility mode
-    mov rsp, STACK_TOP
+    lea rsp, [STACK_TOP]

```

Comment:
Are you sure this is correct? This code does not execute in place, but is copied elsewhere in memory and executed from there, afaict. (in `RelocateMailbox()`)

That might mean this code could be moved into `.rodata`, which might also solve the text relocation issue.